Lib/typing.py

```diff
@@ -1290,13 +1290,30 @@ def __call__(self, *args, **kwargs):
         if not self._inst:
             raise TypeError(f"Type {self._name} cannot be instantiated; "
                             f"use {self.__origin__.__name__}() instead")
-        result = self.__origin__(*args, **kwargs)
+        clas = self.__origin__
+        is_python_type = clas.__new__ is type.__new__
```

Comment:
hmm, this seems wrong. Python types can have custom `__new__` methods:

```pycon
>>> import pathlib
>>> pathlib.Path.__new__ is type.__new__
False
```

Reply:
I guess we can check if it's a native Python function instead.